MAUI/DataGrid/columns.md

```diff
@@ -0,0 +1,205 @@
+documentation: UG
+---
+
+# Columns in MAUI DataGrid (SfDataGrid)
+
+The [SfDataGrid](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html) allows to create and add columns in the following two ways,
+
+* Automatically generating columns
+* Manually defining columns
+
+## Automatic columns generation
+
+The SfDataGrid creates columns automatically based on the bindable property [SfDataGrid.AutoGenerateColumnsMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_AutoGenerateColumnsMode). The columns are generated based on the type of individual properties in the underlying collection that is set as ItemsSource.
+
+Below table shows the column type created for the respective data types. For remaining types, [DataGridTextColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridTextColumn.html) will be created.
+
+<table>
+<tr>
+<th> Data Type </th>
+<th> Column </th>
+</tr>
+<tr>
+<td> string, object </td>
+<td> DataGridTextColumn </td>
+</tr>
+<tr>
+<td> int, float, double, decimal and it’s respective nullable types </td>
+<td> DataGridNumericColumn </td>
+</tr>
+<tr>
+<td> DateTime </td>
+<td> DataGridDateColumn </td>
+</tr>
+<tr>
+<td> bool </td>
+<td> DataGridCheckboxColumn </td>
+</tr>
+<tr>
+<td> ImageSource </td>
+<td> DataGridImageColumn </td>
+</tr>
+</table>
+
+### Different modes to auto generate columns
+
+The auto generation of the columns in `SfDataGrid` is happening based on the `SfDataGrid.AutoGenerateColumnsMode` property. The default value is [AutoGenerateColumnsMode.Reset](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.AutoGenerateColumnsMode.html#Syncfusion_Maui_DataGrid_AutoGenerateColumnsMode_Reset).
+The `SfDataGrid.AutoGenerateColumnsMode` includes the following modes,
+
+<table>
+<tr>
+<th> Modes </th>
+<th> Description </th>
+</tr>
+<tr>
+<td> <code>None</code> </td>
+<td> Stores only the columns that are defined in <code>SfDataGrid.Columns</code> collection.<br/> When changing the <code>ItemsSource</code> and sorting for explicitly defined <code>SfDataGrid.Columns</code> alone will be retained. </td>
+</tr>
+<tr>
+<td> <code>Reset</code> </td>
+<td> Retains the columns defined explicitly in the application level and creates columns newly for all the other properties in a data source.<br/> When changing the <code>ItemsSource</code> and sorting for explicitly defined <code>SfDataGrid.Columns</code> alone will be retained. </td>
+</tr>
+<tr>
+<td> <code>ResetAll</code> </td>
+<td> When changing the <code>ItemsSource</code>, the columns for the previous data source are cleared and the columns will be created newly for the new data source. Even when columns are explicitly defined it does not consider the defined columns and creates the column based on the underlying collection.<br/> Further when changing the <code>ItemsSource</code> and sorting for all the columns will be cleared. </td>

```

Comment:
<td> When changing the <code>ItemsSource</code>, the columns for the previous data source are cleared and the columns will be newly created for the new data source. Even when columns are explicitly defined, it does not consider the defined columns and creates the column based on the underlying collection.<br/> Further, when changing the <code>ItemsSource</code> and sorting for all the columns will be cleared. </td>

Reply:
Modified.